lib/logger/lib/logger/backends/console.ex

```diff
@@ -97,4 +143,41 @@ defmodule Logger.Backends.Console do
   defp color_event(data, level, %{enabled: true} = colors) do
     [IO.ANSI.format_fragment(Map.fetch!(colors, level), true), data | IO.ANSI.reset]
   end
+
+  defp flush_buffer(%{batch_size: 0} = state) do
+    state
+  end
+
+  defp flush_buffer(%{device: device, buffer: buffer} = state) do
+    next_ref = io_request(device, [buffer | dropping(state)])
+    %{state | ref: next_ref, buffer: [], batch_size: 0}
+  end
+
+  defp dropping(%{batch_size: batch_size, max_buffer: max_buffer})
+  when batch_size <= max_buffer do
+    []
+  end
+
+  defp dropping(state) do
+    %{batch_size: batch_size, max_buffer: max_buffer, colors: colors} = state
+    dropped = batch_size - max_buffer
+    msg = "#{inspect __MODULE__} dropped #{inspect dropped} events as it " <>
+          "exceeded the max buffer size of #{inspect max_buffer} messages"
+    color_event(msg, :error, colors)
```

Comment:
I am not super comfortable with this call to color_event. We are now calling it from three different places. Can we create a `format_event` function and call it with this message? The metadata can be empty but we will need some timestamp.